libs/community/langchain_community/chat_models/huggingface.py

```diff
@@ -71,6 +78,37 @@ def validate_llm(cls, values: dict) -> dict:
             )
         return values
 
+    def _stream(
+        self,
+        messages: List[BaseMessage],
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[ChatGenerationChunk]:
+        request = self._to_chat_prompt(messages)
+
+        for data in self.llm._stream(request, **kwargs):
```

Comment:
should this just call self.llm.stream?

Reply:
@baskaryan 

Yes, this is a wrapper for the HuggingFaceTextGenInference, HuggingFace Endpoint, and HuggingFace_Hub models.
So I checked that it is working properly by calling _stream of llm.
For reference, I tested as follows.

(in opengpts.. backend/app/llms.py)

@lru_cache(maxsize=1)
def get_tgi_llm():
    huggingface_hub.login(os.getenv("HUGGINGFACE_TOKEN"))
    llm = HuggingFaceTextGenInference(
        inference_server_url="http://myinferenceserver/",
        max_new_tokens=1024,
        top_k=10,
        top_p=0.95,
        typical_p=0.95,
        temperature=0.3,
        repetition_penalty=1.1,
        streaming=True,
        callbacks=[],
        server_kwargs={
            "headers": {
                "Content-Type": "application/json",
            }
        },
    )
    return ChatHuggingFace(llm=llm, model_id="HuggingFaceH4/zephyr-7b-beta") # setting a model_id for initializing the tokenizer (the another bug)


(in opengpts.. backend/app/agent.py)
def get_agent_executor(
    ...)
    if agent == AgentType.TGI:
        llm = get_gauss_llm()
        system_message = my_prompt
        return get_gauss_agent_executor(
            tools, llm, system_message, interrupt_before_action, CHECKPOINTER
        )
    else:
        raise ValueError("Unexpected agent type")

Reply:
@baskaryan 

After reading the comment you left, I checked not only the class I used, but also the huggingface endpoint and huggingfacehub.
The HuggingFaceEndpoint class had stream-related content, but HuggingFaceHub class only had generate logic without any such streaming logic.
It seems like I need to explicitly specify the type of llm at ChatHuggingFace as Union type in the initialization step, 
but I want to respect the history of changing the type to Any instead of Union.
Instead, I will add streaming support code to the HuggingFaceHub class and check 
whether it works properly and whether it can be integrated with ChatHuggingFace, and then I'll upload the modified code.

Reply:
@eunhye1kim just mean should we use `self.llm.stream` instead of `self.llm._stream`

Reply:
@baskaryan I agree. So I changed the code. In the case of HuggingFaceHub, I think streaming support logic may not be necessary because it is a deprecated class.